src/gc/unix/cgroup.cpp

```diff
@@ -0,0 +1,248 @@
+        {
+            char* context = NULL;
+            char* strTok = strtok_s(options, ",", &context); 
+            while(strTok != NULL)
+            {
+                if(strncmp("memory", strTok, 6) == 0)
+                {
+                    sscanfRet = sscanf_s(line,
+                                         "%*d %*d %*s %*s %s ",
+                                         mountpath, mountpathsize);
+                    if (sscanfRet != 1)
+                    {
+                        _ASSERTE(!"Failed to parse mount info file contents with sscanf_s.");
+                        goto done;
+                    }
+                    retval = true;
+                    break;
+                }
+                strTok = strtok_s(NULL, ",", &context);
+            }
+        }
+    }
+done:
+    if(line)
+        free(line);
+    if(mountinfofile)
+        fclose(mountinfofile);
+    return retval;
+}
+
+bool CGroup::findCGroupPathForMemorySubsystem(char* cgrouppath, size_t cgrouppathsize)
+{
+    bool retval = false;
+    char *line = nullptr;
+    size_t lineLen = 0;
+    FILE *cgroupfile = fopen(PROC_CGROUP_FILENAME, "r");
+    if(cgroupfile == NULL)
+        goto done;
+
+    while(!retval && getline(&line, &lineLen, cgroupfile) != -1)
+    {
+        char subsystem_list[100];
+        char cgroup_path[MAX_PATH];
+
+        // See man page of proc to get format for /proc/self/cgroup file
+        int sscanfRet = sscanf_s(line, 
+                                 "%*[^:]:%[^:]:%s",
+                                 subsystem_list, (unsigned)_countof(subsystem_list),
+                                 cgroup_path, (unsigned)_countof(cgroup_path));
+        if (sscanfRet != 2)
+        {
+            _ASSERTE(!"Failed to parse cgroup info file contents with sscanf_s.");
+            goto done;
+        }
+
+        char* context = NULL;
+        char* strTok = strtok_s(subsystem_list, ",", &context); 
+        while(strTok != NULL)
+        {
+            if(strncmp("memory", strTok, 6) == 0)
+            {
+                strcpy_s(cgrouppath, cgrouppathsize, cgroup_path);
+                retval = true;
+                break;  
+            }
+            strTok = strtok_s(NULL, ",", &context);
+        }
+    }
+done:
+    if(line)
+        free(line);
+    if(cgroupfile)
+        fclose(cgroupfile);
+    return retval;
+}
+
+bool CGroup::readMemoryValueFromFile(const char* filename, size_t& val)
+{
+    bool result = false;
+    char *line = nullptr;
+    size_t lineLen = 0;
+    char* endptr = NULL;
+    size_t num = 0, l, multiplier;
+
+    FILE* file = fopen(filename, "r");
+    if(file == NULL)
+        goto done;
+    
+    if(getline(&line, &lineLen, file) == -1)
+        goto done;
+
+    errno = 0;
+    num = strtoull(line, &endptr, 0); 
+    if(errno != 0)
+        goto done;
+
+    multiplier = 1;
+    switch(*endptr)
+    {
+        case 'g':
+        case 'G': multiplier = 1024;
```

Comment:
yes. thanks for catching it. I did test it before and then changed variable names but missed here.